release_files/install.sh

```diff
@@ -0,0 +1,251 @@
+        SKIP_UI_APP=false
+    fi
+
+    # Identify OS name and default package manager
+    if type uname >/dev/null 2>&1; then
+	case "$(uname)" in
+        Linux)
+            OS_NAME="$(. /etc/os-release && echo "$ID")" 
+            OS_TYPE="linux"
+            INSTALL_DIR="/usr/bin"
+            
+            # Allow netbird UI installation for x64 arch only
+            if [ "$ARCH" != "amd64" ] && [ "$ARCH" != "arm64" ] \
+                && [ "$ARCH" != "x86_64" ];then
+                echo "$ARCH is not a compatible architecture for Netbird UI"
+                SKIP_UI_APP=true
+            fi
+
+            # Allow netbird UI installation for linux running desktop enviroment
+            if [ -z "$XDG_CURRENT_DESKTOP" ];then
+                    SKIP_UI_APP=true
+            fi
+
+            # Check the availability of a compatible package manager
+            if [ -x "$(command -v apt)" ]; then
+                PACKAGE_MANAGER="apt"
+            fi
+            if [ -x "$(command -v yum)" ]; then
+                PACKAGE_MANAGER="yum"
+            fi
+            if [ -x "$(command -v dnf)" ]; then
+                PACKAGE_MANAGER="dnf"
+            fi
+            if [ -x "$(command -v pacman)" ]; then
+                PACKAGE_MANAGER="pacman"
+            fi
+		;;
+		Darwin)
+            OS_NAME="macos"
+			OS_TYPE="darwin"
+            INSTALL_DIR="/usr/local/bin"
+            
+            # Check the availability of a compatible package manager
+            if [ -x "$(command -v brew)" ]; then 
+                PACKAGE_MANAGER="brew"
+            fi
+		;;
+	esac
+    fi
+
+    # Run the installation, if a desktop environment is not detected
+    # only the CLI will be installed
+    case "$PACKAGE_MANAGER" in
+    apt)
+        sudo apt-get update
+        sudo apt-get install ca-certificates gnupg -y
+        
+        curl -sSL https://pkgs.wiretrustee.com/debian/public.key \
+        |  gpg --dearmor --output /usr/share/keyrings/wiretrustee-archive-keyring.gpg
+
+        APT_REPO="deb [signed-by=/usr/share/keyrings/wiretrustee-archive-keyring.gpg] https://pkgs.wiretrustee.com/debian stable main"
+        echo "$APT_REPO" | sudo tee /etc/apt/sources.list.d/wiretrustee.list
+
+        sudo apt-get update
+        sudo apt-get install netbird -y
+        
+        if ! $SKIP_UI_APP; then 
+            sudo apt-get install netbird-ui -y
+        fi
+    ;;
+    yum)
+        add_rpm_repo
+        sudo yum -y install netbird
+        if ! $SKIP_UI_APP; then 
+            sudo yum -y install netbird-ui
+        fi
+    ;;
+    dnf)
+        add_rpm_repo
+        sudo dnf -y install dnf-plugin-config-manager
+        sudo dnf config-manager --add-repo /etc/yum.repos.d/wiretrustee.repo
+        sudo dnf -y install netbird
+
+        if ! $SKIP_UI_APP; then 
+            sudo dnf -y install netbird-ui
+        fi
+    ;;
+    pacman)
+        sudo pacman -Syy
+        sudo pacman -Sy --needed --noconfirm git base-devel go
```

Comment:
I got this error on arch:

```shell
warning: Public keyring not found; have you run 'pacman-key --init'?
downloading required keys...
error: keyring is not writable
error: keyring is not writable
error: keyring is not writable
error: keyring is not writable
error: keyring is not writable
error: keyring is not writable
error: keyring is not writable
error: keyring is not writable
error: keyring is not writable
error: required key missing from keyring
error: failed to commit transaction (unexpected error)
Errors occurred, no packages were upgraded.
```

Reply:
I have tested it on Manjaro and encountered no problems. It is possible that the issue you are experiencing is due to the Pacman keyring being absent.